pkg/sops/sops.go

```diff
@@ -0,0 +1,88 @@
+package sops
+
+import (
+	"fmt"
+	"io/fs"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"github.com/getsops/sops/v3"
+	"github.com/getsops/sops/v3/stores/dotenv"
+	"github.com/getsops/sops/v3/stores/ini"
+	"github.com/getsops/sops/v3/stores/json"
+	"github.com/getsops/sops/v3/stores/yaml"
+)
+
+func getStore(fileType string) (sops.Store, error) {
+	switch fileType {
+	case ".ini":
+		return &ini.Store{}, nil
+	case ".env":
+		return &dotenv.Store{}, nil
+	case ".yaml", ".yml":
+		return &yaml.Store{}, nil
+	case ".json":
+		return &json.Store{}, nil
+	default:
+		return nil, fmt.Errorf("Unsupported file type: %s", fileType)
+	}
+}
+
+// File represents a SOPS file and its metadata
+type File struct {
+	Path     string
+	Metadata sops.Metadata
+}
+
+// FindFiles searches a directory for YAML files and checks if they are valid SOPS files.
+// Returns []SopsFile
+func FindFiles(root string) ([]File, error) {
+	var SopsFiles []File
+
+	err := filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Skip directories
+		if d.IsDir() {
+			return nil
+		}
+
+		ext := filepath.Ext(d.Name())
+
+		store, err := getStore(ext)
+		if err != nil {
+			return nil
+		}
+
+		encrypteddata, err := os.ReadFile(path)
```

Comment:
Variable names should use camelCase:

```suggestion
		encryptedData, err := os.ReadFile(path)
```

I'd prefer to use `data` here though, because it's more correct. After reading the file we don't know yet if it actually contains encrypted data or not.